module/User/src/Controller/ApiSettingsController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace User\Controller;
+
+use Laminas\Http\Response;
+use Laminas\Mvc\Controller\AbstractActionController;
+use Laminas\Mvc\Plugin\FlashMessenger\FlashMessenger;
+use Laminas\View\Model\ViewModel;
+use User\Service\ApiPrincipalService;
+
+/**
+ * @method FlashMessenger flashMessenger()
+ */
+class ApiSettingsController extends AbstractActionController
+{
+    /**
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification
+     */
+    public function __construct(
+        protected readonly ApiPrincipalService $apiPrincipalService,
+        protected readonly array $config,
+    ) {
+    }
+
+    /**
+     * View API principals.
+     */
+    public function listPrincipalsAction(): ViewModel
+    {
+        return new ViewModel([
+            'principals' => $this->apiPrincipalService->findAll(),
+        ]);
+    }
+
+    /**
+     * Create an API principal
+     */
+    public function createPrincipalAction(): Response|ViewModel
+    {
+        $form = $this->apiPrincipalService->getCreateForm();
+
+        if ($this->getRequest()->isPost()) {
+            $result = $this->apiPrincipalService->create(
+                $this->getRequest()->getPost()->toArray(),
+                $this->flashMessenger(),
+            );
```

Comment:
Instead of returning `true` I would return the `ApiPrincipal` and prevent having to pass the `flashMessenger`.

Reply:
b48bbb2e9966b14482d8bd2b22ee043892724aac